### DISCUSSION TITLE
Would a native Vulkan driver written for M2's GPU provide performance benefits over MoltenVK?

### DISCUSSION THREAD

**Discussion by scheeseman486** [KhronosGroup/MoltenVK#1933] [General]
Given Apple are feeling generous would there be a benefit to a "native" Vulkan driver? Particularly one that doesn't wrap through Metal on OSX? Based on my conversations with an outside developer they seem to be sure that whatever overhead that exists to translate between immediate rendering to tile based deferred rendering can't be eliminated by going lower level and it wouldn't be worth it. Are there private APIs or hardware registers that'd help close the gap?

Question is also somewhat relevant to Asahi Linux, since presumably that would also get hit by the same performance degradation.

**Comment by billhollings** [comment]
> Given Apple are feeling generous would there be a benefit to a "native" Vulkan driver? Particularly one that doesn't wrap through Metal on OSX?

Can you expand on what you are proposing here? What are you suggesting Apple is feeling generous about? And how are you thinking a native Vulkan driver would be accomplished?

**Reply by scheeseman486** [reply]
Not really a proposal, just a question about whether there would be performance or compatibility benefits given a a Vulkan driver for Apple Silicon on OSX compared to the approach MoltenVK takes, irrespective of Apple's design philosophies (violating these would be "generous" of them) and the lack of any way to do such a thing without their involvement.

I posted on Reddit that native support for Vulkan would be of benefit and it got a ton of upvotes, like, hundreds. Not tooting my own horn, it actually kind of bothers me since there were responses that made a semi-convincing case that I was wrong that were subsequently downvoted. I mostly just want to know the truth, I'd rather not be spreading falsehoods.

**Reply by billhollings** [reply]
A native Vulkan driver is definitely every game dev's preferred best option, and many have requested it. That's why you're seeing so many upvotes. But the only people that can make that happen are Apple, and so far, they haven't expressed an interest in it, because their focus is on Metal.
